app/helpers/users_helper.rb

```diff
@@ -41,4 +43,12 @@ def user_has_open_application(user)
     user.shf_application.state.to_sym.in?([:accepted, :rejected]) ? nil : t('yes')
   end
 
+  def short_proof_of_membership_url(user)
+    TinyURL.short(proof_of_membership_url(user))
+  end
+
+  def short_company_h_brand_url(user, company_id)
+    TinyURL.short(company_h_brand_url(user, company_id: company_id))
+  end
```

Comment:
These two methods should be Company instance methods (hence, defined in the model file).

These methods should first determine if we already have a shortened URL for the link - if so, just return that.  If not, call the service class and save the result if not `nil`.  

Finally, if we don't have a shortened URL for the link (because it's not in the DB and the service call fails), then just return the unshortened URL.

Reply:
Consistent with above comment, we need a migration to add the shortened URL's to the DB.